Card/Masked.ts

```diff
@@ -9,14 +11,31 @@ export interface Masked {
 	lapses?: isoly.Date
 }
 export namespace Masked {
-	export function is(value: Masked | any): value is Masked {
-		return (
-			typeof value == "object" &&
-			typeof value.masked == "string" &&
-			typeof value.iin == "string" &&
-			typeof value.last4 == "string" &&
-			Expires.is(value.expires) &&
-			(value.lapses == undefined || isoly.Date.is(value.lapses))
-		)
+	export const type = isly.object<Masked>(
+		{
+			masked: isly.string(/^\d{6,8}\*+\d{4}$/),
+			iin: isly.string(/^\d{6,8}$/),
+			last4: isly.string(/^\d{4}$/),
+			expires: Expires.type,
+			lapses: isly.fromIs<isoly.Date>("isoly.Date", isoly.Date.is).optional(),
+		},
+		"pax2pay.cde.Card.Masked"
+	)
+	export const is = type.is
+	export const flaw = type.flaw
+	export function mask(card: Card): Masked {
+		const length = card.pan.length
+		const iinLength = ["45561800", "45561900", "49299300", "49299400"].some(iin => card.pan.startsWith(iin)) ? 8 : 6
+		const iin = card.pan.slice(0, iinLength)
+		const last4 = card.pan.slice(length - 4, length)
+		const result: Masked = {
+			masked: iin + "*".repeat(length - iinLength + 4) + last4,
```

Comment:
Shouldn't it be `length - iinLength - 4`?

Reply:
https://github.com/pax2pay/model-cde/pull/51/files#diff-4f8b99d8b8fcac8607c125cfb80a54df8dda3bb4e84b89cd020c7a9f1137422cL46

this function is mostly just moved from elsewhere

Reply:
I think you're correct, this field hasn't been used that much so it has likely just been a hidden bug. I'll change it